### ISSUE TITLE
[4.x]: GraphQL filter on siteId throws Invalid site param

### ISSUE THREAD

**Issue by roland-d** [craftcms/cms#13340]
### Description
I am testing a GraphQL query in the GraphQL Explorer and to limit my results I am filtering on either `site` or `siteId`. First I tried to filter on `site` like this `site:["SITE_NAME"]`. This results in the error `Invalid site param:`. So I tred to filter on `siteId` like this: `siteId:1` or `siteId:["1"]` but this returns an empty result set. Finally I removed the site filter to just get all records and all records are returned and I can see the siteId value as well in the result set.

### Steps to reproduce

1. Use a Craft site with at least 2 sites
2. Create a GraphQL Schema
3. Check the `Entries`
4. Make sure the `Sites` are all unchecked
5. Save the schema
6. Run a GraphQL query on the site with a `site` or `siteId` filter
7. Notice either an error or an empty result set
8. Run the query again but this time without a filter
9. Notice that all records are returned

### Expected behavior
When I run the GraphQL query I expect no results when not using a site filter because no site has been checked in the GraphQL schema. This would otherwise reveal data that should only be available when a site has been checked.

### Actual behavior
All entries are returned when not filtering on a site

### Craft CMS version

4.4.14

### PHP version

8.2.7

### Operating system and version

Linux

### Database type and version

MySQL 5.7.8

### Installed plugins and versions

- Redactor | 3.0.x-dev

**Comment by i-just** [comment]
Hi, thanks for reporting! I raised a PR to enforce site access based on the used schema when querying without providing the `site` or `siteId` filters.

@kbergha, are you using the same schema in all the cases you mentioned?

**Comment by kbergha** [comment]
@i-just - Yes. To be specific, it's the default public schema:

![Firefox_Screenshot_2023-06-21T18-12-54 316Z](https://github.com/craftcms/cms/assets/4321846/dc622980-75ff-4d8a-adff-070af136e9d8)
![Firefox_Screenshot_2023-06-21T18-11-07 261Z](https://github.com/craftcms/cms/assets/4321846/182ad051-3815-43ab-a840-39b50dba0ea8)

**Comment by i-just** [comment]
Thanks for clarifying, @kbergha!

The first thing to note is that you should select that one site in the schema. The idea is: it defines which site(s) can have their elements queried in the first place. You then choose which elements should be queryable (in your case, that’s “Artikkel” entries, “Filer” assets and all categories). Without selecting any site, no elements should be queryable. That being said, there’s currently a bug that will be fixed when the PR I raised concerning this issue is released.

>Edit: I see I selected the "Full schema" in the GraphQL explorer in my previous message.
Trying again on 4.4.14 with the public schema actually selected - no records returned, so at least it's consistent :)

Based on my testing, the query is consistent between `4.4.6.1` and `4.4.14`. I get zero results with the `siteId` filter (because the schema doesn’t have access to any sites) and all results without it (because of a bug). 
Can you please check if the Public Schema in your `4.4.6.1` installation has that one site selected?

**Comment by kbergha** [comment]
The public schema does not have a site selected in 4.4.6.1.
You can try the query against our production env. at https://aktuelt.oslo.kommune.no/api if you want.

This is the content of the YAML that's in production on 4.4.6.1 right now (no site reference):
```
isPublic: true
name: 'Public Schema'
scope:
  - 'sections.80c290a1-777f-4f13-af2d-3789bc6e7942:read' # Artikler
  - 'entrytypes.d7475be8-537e-4ff1-b081-b2d80af3008c:read' # Artikkel
  - 'volumes.0bf32112-2a77-4641-b36a-ec937c48c5cc:read' # Filer
  - 'categorygroups.ff71f8bf-79fd-4885-976b-724792c5c044:read' # Annen tilhørighet
  - 'categorygroups.c9df9d82-d5ec-4e37-a8bd-8f3519dd00bd:read' # Etater, foretak og bydeler
  - 'categorygroups.2f08e643-1f44-400b-af6b-79ef53e746a5:read' # Byrådsavdelingene
  - 'categorygroups.0f305cb9-1025-4777-b9c6-3e1a8508e586:read' # Tema
  - 'categorygroups.ae68e57b-f625-4966-83cf-71c6882f7e75:read' # Type
```

The query with the siteId filter works if I in 4.4.14 add the site as you mentioned. 
The YAML then looks like:
```
isPublic: true
name: ‘Public Schema’
scope:
  - ‘sites.f420e2ae-0c31-4738-9df0-581c70353f9d:read’ # Oslo kommune - Aktuelt
  - ‘sections.80c290a1-777f-4f13-af2d-3789bc6e7942:read’ # Artikler
  - ‘entrytypes.d7475be8-537e-4ff1-b081-b2d80af3008c:read’ # Artikkel
  - ‘volumes.0bf32112-2a77-4641-b36a-ec937c48c5cc:read’ # Filer
  - ‘categorygroups.ff71f8bf-79fd-4885-976b-724792c5c044:read’ # Annen tilhørighet
  - ‘categorygroups.2f08e643-1f44-400b-af6b-79ef53e746a5:read’ # Byrådsavdelingene
  - ‘categorygroups.c9df9d82-d5ec-4e37-a8bd-8f3519dd00bd:read’ # Etater, foretak og bydeler
  - ‘categorygroups.0f305cb9-1025-4777-b9c6-3e1a8508e586:read’ # Tema
  - ‘categorygroups.ae68e57b-f625-4966-83cf-71c6882f7e75:read’ # Type
```

I understand I can fix this by adding the missing site to the public schema.
Should my query with the site id filer not have worked in the first place?

**Comment by kbergha** [comment]
By the way, I think the reason for not selecting the site in the first place might be the wording of the headings/labels on the selection page.

The label for the site mentions elements, then the next heading says "All elements", and then refers to drafts, revisions++. 

My probable thinking when reading this the first time:
What's the difference between site elements and all elements? I don't want to allow access to drafts and revisions, ignore that! Ah, entries, assets and categories, that's what it need.

![Firefox_Screenshot_2023-06-22T09-46-58 367Z](https://github.com/craftcms/cms/assets/4321846/7fe200f7-6e65-4caf-981a-056e1190cd1b)

**Comment by kbergha** [comment]
Thank you!

**Comment by roland-d** [comment]
@i-just I do not think my issue is addressed yet. In your recap you said:
> All elements belong to site(s) - that’s why the site(s) you want to be able to query should be selected.

In my multi-site setup, if I do not select any site, all entries are returned. Is the PR you raised addressing this? If so, then my question is answered.

**Comment by i-just** [comment]
> Is the PR you raised addressing this?

Yes, it is :)

**Comment by brandonkelly** [comment]
The PR has been merged, however I targeted the 4.5 branch for it, as there’s a chance it could introduce an unexpected change in behavior for GraphQL schemas that don’t have a site enabled, and are only getting access to the desired site’s content accidentally via this bug.

**Comment by rrr3da** [comment]
Hi guys,

We run an update from 4.4.16.1 to 4.5.5 on a single-site configuration, and GraphQL was fixed by flagging the **"Allow querying elements from ...."**

Thank you for your hints 👍
